packages/xo-server-sdn-controller/src/index.js

```
@@ -0,0 +1,821 @@
+import assert from 'assert'
+import createLogger from '@xen-orchestra/log'
+import NodeOpenssl from 'node-openssl-cert'
+import { access, constants, readFile, writeFile } from 'fs'
+import { EventEmitter } from 'events'
+import { filter, find, forOwn, map } from 'lodash'
+import { fromCallback, fromEvent } from 'promise-toolbox'
+import { join } from 'path'
+
+import { OvsdbClient } from './ovsdb-client'
+
+// =============================================================================
+
+const log = createLogger('xo:xo-server:sdn-controller')
+
+const PROTOCOL = 'pssl'
+
+const CA_CERT = 'ca-cert.pem'
+const CLIENT_KEY = 'client-key.pem'
+const CLIENT_CERT = 'client-cert.pem'
+
+const SDN_CONTROLLER_CERT = 'sdn-controller-ca.pem'
+
+const NB_DAYS = 9999
+
+// =============================================================================
+
+export const configurationSchema = {
+  type: 'object',
+  properties: {
+    'cert-dir': {
+      description: `Full path to a directory where to find: \`client-cert.pem\`,
+ \`client-key.pem\` and \`ca-cert.pem\` to create ssl connections with hosts.
+ If none is provided, the plugin will create its own self-signed certificates.`,
+
+      type: 'string',
+    },
+    'override-certs': {
+      description: `Replace already existing SDN controller CA certificate`,
+
+      type: 'boolean',
+      default: false,
+    },
+  },
+}
+
+// =============================================================================
+
+async function fileWrite(path, data) {
+  try {
+    await fromCallback(writeFile, path, data)
+    log.debug(`${path} successfully written`)
+  } catch (error) {
+    log.error(`Couldn't write in: ${path} because: ${error}`)
```

Is it a good idea to hide the error?